docs/source/upgrade/upgrade-guide-from-monitoring-3.x-to-monitoring-4.y.rst

```diff
@@ -0,0 +1,257 @@
+
+.. code-block:: bash
+
+                ./start-all.sh --version
+
+To validate the Scylla-Monitoring version.
+
+
+Validate the version installed correctly
+----------------------------------------
+
+To validate that the Monitoring stack starts correctly, first in parallel to the current (3.x) stack.
+
+.. code-block:: bash
+
+                ./start-all.sh -p 9091 -g 3001 -m 9095
+
+Browse to ``http://{ip}:9091``
+And check the Grafana dashboard
+
+Note that we are using different port numbers for Grafana, Prometheus, and the Alertmanager.
+
+.. caution::
+
+   Important: do not use the local dir flag when testing!
+
+When you are satisfied with the data in the dashboard, you can shut down the containers.
+
+.. caution::
+
+   Important: Do not kill the 3.x version that is currently running.
+
+Killing the new 4.y Monitoring stack in testing mode
+----------------------------------------------------
+
+Use the following command to kill the containers:
+
+.. code-block:: bash
+
+                ./kill-all.sh -p 9091 -g 3001 -m 9095
+
+You can start and stop the new 3.y version while testing.
+
+Move to version 4.y (the new version)
+-------------------------------------
+
+Note: migrating will cause a few seconds of blackout in the system.
+
+We assume that you are using external volume to store the metrics data.
+
+Kill all containers
+^^^^^^^^^^^^^^^^^^^
+
+At this point you have two monitoring stacks running side by side, you should kill both before
+continuing.
+
+Kill the newer version that runs in testing mode by following the instructions on how to `Killing the new 4.y Monitoring stack in testing mode`_
+in the previous section
+
+kill the older 3.x version containers by running:
+
+.. code-block:: bash
+
+                ./kill-all.sh
+
+Start version 4.y in normal mode
+
+
+From the new root of the `scylla-monitoring-scylla-monitoring-4.y` run
+
+.. code-block:: bash
+
+                ./start-all.sh -d /path/to/data/dir
+
+
+Point your browser to ``http://{ip}:3000`` and see that the data is there.
+
+Rollback to version 3.x
+-----------------------
+
+
+To rollback during the testing mode, follow `Killing the new 4.y Monitoring stack in testing mode`_ as explained previously
+and the system will continue to operate normally.
+
+To rollback to version 3.x after you completed the moving to version 4.y (as shown above).
+Run:
+
+.. code-block:: bash
+
+                ./kill-all.sh
+                cd /path/to/scylla-grafana-3.x/
+                ./start-all.sh -d /path/to/data/dir
+
+Post-installation: Metrics back-filling
+=======================================
+Starting from Scylla Monitoring version 3.8, Scylla Monitoring uses Prometheus's recording rules for performance reasons.
+Recording rules perform some of the calculations when collecting the metrics instead of when showing the dashboards.
+
+For a transition period, Scylla Monitoring version 3.x has a fall-back mechanism that if those recording rules
```

Comment:
is it all 3.x or just 3.8, 3.9?

Reply:
all versions prior to 4.0 does not relay solly on recording rules (before 3.8 there are no recording rules)